### PR TITLE
feat(nd): VOR/ADF needles on ARC screen

### LINKED ISSUE #3055: [REQUEST] Navigation Needles in ARC Modes
As in the real Airbus there have to be two needles for the VOR in the ARC Mode.

**Is your feature request related to a problem? Please describe.**

No

**Describe the feature you'd like to see implemented**

Needle in ARC Mode

**References**

Every Airbus

**Additional context**

Discord username (if different from GitHub):

**BY REMOVING THIS LINE, I ATTEST THAT I HAVE FOLLOWED THE ISSUE TEMPLATE AND PROVIDED A TITLE, as well as searched for other possible duplicate issues. I acknowledge if this line is not removed and/or the issue template is not respected, my issue will be closed without warning**

### PR DESCRIPTION
**Pull request by tracernz**
Fixes #3055

To be sequenced after #3884 

## Summary of Changes

- Show VOR/ADF needles on ARC page
- Hide VOR/ADF needles on ILS page (I don't think they're supposed to be there looking at FCOM)
- Split control of needles so captain's are controlled by switches on captain's side and vice-versa.

## Screenshots (if necessary)

image

## References

image

## Additional context

I took the needle SVG path from Asobo's latest patch. If there are any budding SVG artists they could probably be improved.

Discord username (if different from GitHub):

## Testing instructions

Tune ADFs and/or VORs:
- Turn on needles
- Check that needles are displayed on all pages except LS and PLAN
- Check that needles point in direction of tuned navaids
- Check that switches on EFIS panel control ND needles on same side of the cockpit
- Check that ADF needles are green and VOR needles are white

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by sidnov** [comment]
Awesome!🎉 Will reopen my PR once this is implemented 😀

**Comment by tracernz** [comment]
> Awesome!🎉 Will reopen my PR once this is implemented 😀

I don't think yours should conflict/they should both merge cleanly. :)

**Comment by tracernz** [comment]
Rebased after #3884

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js [hunk 1]
```diff
@@ -447,38 +447,39 @@ class Jet_NDCompass extends HTMLElement {
...
-                            displayVerticalDeviation = true;
-                            this.setAttribute("vertical_deviation", (SimVar.GetSimVarValue("NAV GSI:" + beacon.id, "number") / 127.0).toString());
-                        }
-                    } else {
-                        this.setAttribute("course", compass.toString());
-                        this.setAttribute("course_deviation", "0");
+            const compass = Number(this.getAttribute('rotation'));
+            let displayCourseDeviation = false;
+            let displayVerticalDeviation = false;
+            if (this.navigationMode == Jet_NDCompass_Navigation.ILS || this.navigationMode === Jet_NDCompass_Navigation.VOR) {
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js:453]: can we use strict comparison over `==` ?

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js [hunk 2]
```diff
@@ -447,38 +447,39 @@ class Jet_NDCompass extends HTMLElement {
...
+                let beacon;
+                if (this.navigationMode == Jet_NDCompass_Navigation.ILS) {
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js:455]: same here

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js [hunk 3]
```diff
@@ -447,38 +447,39 @@ class Jet_NDCompass extends HTMLElement {
...
+                    beacon = this.gps.radioNav.getBestILSBeacon();
+                } else if (this.navigationMode === Jet_NDCompass_Navigation.VOR) {
+                    beacon = this.gps.radioNav.getBestVORBeacon();
+                }
+                if (beacon.id > 0) {
+                    displayCourseDeviation = true;
+                    let deviation = (SimVar.GetSimVarValue("NAV CDI:" + beacon.id, "number") / 127);
+                    const backCourse = SimVar.GetSimVarValue("AUTOPILOT BACKCOURSE HOLD", "bool");
+                    if (backCourse) {
+                        deviation = -deviation;
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js:465]: I don't really understand this part.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js [hunk 4]
```diff
@@ -447,38 +447,39 @@ class Jet_NDCompass extends HTMLElement {
...
+                } else {
+                    this.setAttribute("course", compass.toString());
+                    this.setAttribute("course_deviation", "0");
                 }
+            }
+            if (this.courseDeviation) {
                 this.setAttribute("display_course_deviation", displayCourseDeviation ? "True" : "False");
                 this.setAttribute("display_vertical_deviation", displayVerticalDeviation ? "True" : "False");
+            }
+            if (this.navigationMode != Jet_NDCompass_Navigation.ILS) {
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js:482]: same here

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A32NX_NDCompass.js [hunk 5]
```diff
@@ -156,6 +146,49 @@ class Jet_MFD_NDCompass extends Jet_NDCompass {
                         graduationGroup.appendChild(line);
                     }
                 }
+                this.courseGroup = document.createElementNS(Avionics.SVG.NS, "g");
+                this.courseGroup.setAttribute("id", "CourseInfo");
+                this.rotatingCircle.appendChild(this.courseGroup);
+                {
```
Comment (MisterChocker) [inline A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A32NX_NDCompass.js:152]: why do we need separate block here?

**Review by MisterChocker — CHANGES_REQUESTED** [review]

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js (hunk 3 above)
Comment (tracernz) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js:465]: Is Asobo code from the changes they made in SU3... I don't know what the deviation does so I didn't alter their code.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js (hunk 1 above)
Comment (tracernz) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/A32NX_BaseNDCompass.js:453]: Sure. 👍

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A32NX_NDCompass.js (hunk 5 above)
Comment (tracernz) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A32NX_NDCompass.js:152]: Happy to drop that, although it follows the style of all the other code around if you scroll down a bit.

**Review by beheh — APPROVED** [review]
I think this is mostly fine as-is until we do a lot more work on the ND. The script comparisons that have been called out would be good to change, I don't particularly mind the blocks.

File: A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A32NX_NDCompass.js (hunk 5 above)
Comment (MisterChocker) [reply A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/MFD/A32NX_NDCompass.js:152]: yeah, okay to keep them.

**Comment by tracernz** [comment]
Thanks @MisterChocker and @beheh. The comparisons are now fixed and the PR rebased.

**Review by MisterChocker — APPROVED** [review]
LGTM

**Comment by marcman86** [comment]
Category: QA Tester
Name: marcman86#4907
Date of testing: 19.3.2021
Version of the sim: 1.14.5.0
PR Tested: #3886
Tier of testing: 1
Changes to observe: VOR/ADF needles on ARC
Testing technique: Tryout

Steps made to test the Pull Request:

1.) Spawned on runway at EDDP
2.) Took off
3.) Set VOR1 MAG, VOR2 LEG, ADF1 ZIG and ADF2 ZIG
4.) Tested needle drawing by flying some circles, different ARC ranges and switches
5.) Tested signal loss by overflying a VOR transponder

Issues: -

Overall Rating: Good

Conclusions of the testing: Works

Hope this helps :-)

**Comment by St54Kevin** [comment]
Category: QA Tester
Name: St54Kevin#3583
Date of testing: 19/03/2021
Version of the sim: 1.14.5.0
PR Tested: #3886 
Tier of testing: 1
Changes to observe: VOR/NDB needles on ND

**Steps you've made to test the Pull Request:**
1. Loaded onto RW07R in EDDF
2. Tuned in FFM and FW and tested all displays on each pilot side
3. Tested different switching of NAV1/2

**Media:**

**Issues (If Any):** -

**Conclusions of the testing:** LGTM. Thanks for this addition. I really like it and it works as it should.

**Comment by sidnov** [comment]
Category: QA Tester
Name: sidnov#8337
Date of testing: 19/03/2021
Version of the sim: 1.14.5.0
PR Tested: #3886
Tier of testing: 1

### Changes to observe: VOR/NDB needles on ARC

### Steps you've made to test the Pull Request:

1. Spawned C/D, prepped for routine IFR hop
2. Tuned origin VOR, and first enroute VOR
3. Departed, and observed the needles on the ARC screen
4. Switched between different VORs and observed the needle movements along with the aircraft navigating

### Notes:
VOMM KAMGU W20 LEDOB VOHS
Weather: default
Navdata: Navigraph Beta MSFS

### Issues (If Any): None

### Overall Rating: Good

### Conclusions of the testing:
The VOR needles showed up on the ARC display, and function as expected. The controls of Captain/FO's ND needles are also split.

**Pull request merged**